common/src/java/org/apache/hadoop/hive/common/type/Date.java

```diff
@@ -175,22 +178,16 @@ public void setTimeInMillis(long epochMilli) {
    */
   public static Date valueOf(final String text) {
     String s = Objects.requireNonNull(text).trim();
-    int idx = s.indexOf(" ");
-    if (idx != -1) {
-      s = s.substring(0, idx);
-    } else {
-      idx = s.indexOf('T');
-      if (idx != -1) {
-        s = s.substring(0, idx);
-      }
-    }
-    LocalDate localDate;
+    ParsePosition pos = new ParsePosition(0);
     try {
-      localDate = LocalDate.parse(s, PARSE_FORMATTER);
+      TemporalAccessor t = PARSE_FORMATTER.parseUnresolved(s, pos);
```

Comment:
The DateTimeFormatter::parseUnresolved() documentation mentions that it is meant for advanced use cases and that typical applications should use the parse() method.  I suppose that since you are accessing the internal state of the TemporalAccessor, it does match such a 'advanced' use case. 
It would be useful to add a comment here about why parseUnresolved() is used. Further, it would be useful to include in the comment how the validation is done since this method does not do validation of the actual value (e.g if the month is 15).  The validation is deferred to the later call LocalDate.of() method.

Reply:
I was thinking that the commit message was enough to understand why `parseUnresolved` is used instead of `parse`.
Usually, I tend to add comments when the code is not easy to understand not to justify why a change was done. I tend to document the "why part" in the commit message/JIRA but this is an entirely personal contribution style.
Anyways more is usually better than less so I will add an appropriate comment.

Reply:
I am good with a brief comment since the word 'Unresolved' in the api makes one pause and think how the resolution is happening.